packages/mongoose/src/createMongooseController.ts

```diff
@@ -184,7 +185,9 @@ export const createMongooseController = <T extends Constructor<{}>>(
 
 			const [data, total] = await bluebird.all([
 				populate ? mQuery.populate(populate) : mQuery,
-				this.model.countDocuments(where).setOptions({ context: ctx })
+				(_.isEmpty(where) && config.ALLOW_ESTIMATED_DOCUMENT_COUNT) ?
```

Comment:
remove the conditional:
```ts
useEstimatedDocumentCount 
  ? this.model.estimatedDocumentCount(where).setOptions({ context: ctx })
  : this.model.countDocuments(where).setOptions({ context: ctx })
```

Reply:
I am not sure about removing the conditional, as we may want to use the real countDocuments when there is a where clause (that predicate will be used in the query as well, so it should not be a problem). The problem we faced was when the where was empty object as it was doing a full collection scan

Reply:
also the estimatedDocumentCount does not accept a where clause (@adriasala124 updated the code) so to take the filter into account we need to use the real countDocuments instead

Reply:
@juanjoseruiz ok that's fine, let's keep the conditional.
But let's use the estimatedDocumentCount (for queries with no 'where') as the default.

As a separate note, estimatedDocumentCount [seems to accept parameters](https://mongoosejs.com/docs/api.html#model_Model.estimatedDocumentCount)  :thinking:

Reply:
@juanjoseruiz found in the [docs](https://www.mongodb.com/docs/manual/reference/method/db.collection.estimatedDocumentCount/#mechanics) that is not using them @pierissimo

Reply:
ah right, but those are the mongodb docs, we should probably be looking at the mongoose docs

Reply:
Right! Still in the mongoose docs it says it accepts options, but not a filter. I think we cannot pass it the where there.

Reply:
Yep, I think mongoose doc for Model.estimatedDocumentCount is misleading. As it refers to the model, it returns the estimated count of all the documents in the collection.

If you look at estimatedDocumentCount for queres [here](https://mongoosejs.com/docs/api/query.html#query_Query-estimatedDocumentCount) is explicitly says the same as the mongodb driver doc

Reply:
Got you, fixing the mongoose doc would probably be a good OSS contribution :)